src/parser.y

```diff
@@ -184,15 +184,16 @@ statement
         | '{' inner_input '}'
         | module_instantiation
             {
-              if ($1) scope_stack.top()->addChild($1);
+              if ($1) scope_stack.top()->addChild(std::move(shared_ptr<ModuleInstantiation>($1)));
```

Comment:
I don't think std::move() does anything here. The called function does perform the actual move.

Reply:
The called function takes shared_ptr by value, so to move it the value out of parser.y without copying I thought this was necessary?  Are you sure it would still do the move automatically?  

As far as I understand, each call to a function taking a shared_ptr by value would need to use std::move to avoid copy by value.

I could also change the `LocalScope::add*` function signatures to take `shared_ptr<...>&&` arguments which I think should be functionally equivalent with maybe a bit clearer interface, but I think the move would still be required at each call site to relinquish its value to the callee?

Reply:
Nevermind, resolved in IRC conversation